### PR TITLE
Blog app - Add formsForms

### PR DESCRIPTION
**Pull request by VaneCode**
In this project I implemented the following requirements:

- Created new routes to handle forms' actions.
- Added methods new and create in PostsController.
- Added methods create in CommentsController and LikesController.
- Created the necessary views and partials to show the forms.
- Allowed the user to create a post, add a comment, and make a like.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,4 +7,21 @@ def show
...
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @author = current_user
+    @post = Post.new(params.require(:post).permit(:title, :text))
```
Comment (mmsesay) [inline app/controllers/posts_controller.rb]: - [OPTIONAL] This is good 👍 But do you know that you can improve on the code by using [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

```ruby
private 
def post_params
    params.require(:post).permit(:title, :text)
end
```

That private method can be now pass as the new params
```ruby
@post = @current_user.posts.new(post_params)
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    if @comment.save
+      flash[:success] = 'Comment saved successfully.'
+      redirect_to user_post_path(@p_author, @post)
+    else
+      flash.now[:error] = 'Error: Comment could not be saved.'
+      render :new
+    end
+  end
+end
```
Comment (mmsesay) [inline app/controllers/comments_controller.rb]: - [ ] The comments controller should have new and create actions. Only the create is existing.

```ruby
def new
    @comment = Comment.new
end
```

- [OPTIONAL] You can make use of the strong parameters that I suggested in my previous comment.

```ruby
private 
def comment_params
   params.require(:comment).permit(:text)
end
```

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @VaneCode, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title :heavy_check_mark: 
 - Good PR summary ✔️   


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jaflih — APPROVED** [review]
Hi @VaneCode  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

Good Readme ✔️
No linter error ✔️
Good PR title ✔️
Good PR summary ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @jaflih in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**